src/lib/server/subscriptionUtils.ts

```diff
@@ -140,7 +142,8 @@ export const createStripeSession = async (
       },
     ],
     subscription_data: subscriptionData,
-    allow_promotion_codes: true,
+    allow_promotion_codes: promoCode ? undefined : true,
+    discounts: promoCode ? [{ promotion_code: promoCode.id }] : undefined,
```

Comment:
Stripe does not let you define both of these fields, even if I pass `false` to `allow_promotion_codes` when `promoCode` is defined, so instead I have to make them explicitly undefined.

Reply:
weird, but okay makes sense